app/scripts/controllers/plugins.js

```diff
@@ -1,10 +1,19 @@
 const ObservableStore = require('obs-store')
 const EventEmitter = require('safe-event-emitter')
 const extend = require('xtend')
+const { ethErrors, serializeError } = require('eth-json-rpc-errors')
+
 const {
   pluginRestrictedMethodDescriptions,
 } = require('./permissions/restrictedMethods')
-const { ethErrors } = require('eth-json-rpc-errors')
+const { PLUGIN_PREFIX } = require('./permissions/enums')
+
+const ENUMS = {
+  excludeFromSerialization: [
```

Comment:
This is a nice addition. I suspect a permit-list would be slightly easier to securely maintain, since we couldn't forget forget to serialize some functional data, but we could easily forget to remove some sensitive data.

Reply:
Changed to an `Object<string, boolean>` called `shouldSerialize`.